src/web/components/elements/YoutubeBlockComponent.tsx

```diff
@@ -76,6 +78,35 @@ export const YoutubeBlockComponent = ({
     duration,
     origin,
 }: Props) => {
+    const wrapperRef = useRef<HTMLDivElement | null>(null);
+    const [selectedPosterImage, setSelectedPosterImage] = useState<
+        PosterImageType
+    >();
+
+    // We need to find the best candidate posterImage from array
+    // we therefore need to find the candidate with the smallest
+    // width, that is larger than the width of the wrapper
+    useEffect(() => {
+        const wrapperWidth = wrapperRef.current
+            ? wrapperRef.current.clientWidth
+            : 0;
+
+        if (wrapperWidth && posterImage) {
+            const bestFitPosterImage = posterImage.reduce(
+                (
+                    acc: PosterImageType,
+                    cur: PosterImageType,
+                ): PosterImageType => {
+                    if (wrapperWidth < cur.width && cur.width < acc.width)
+                        return cur;
+                    return acc;
+                },
+                posterImage[0],
+            );
+            setSelectedPosterImage(bestFitPosterImage);
+        }
+    }, [wrapperRef, posterImage, setSelectedPosterImage]);
```

Comment:
I'm not an expert on this but does the `picture` tag or maybr the `Image` component perhaps already do this?

Reply:
yes it does!!! Ill split it out into `lib`. Great catch!

Reply:
I was more thinking how the `Picture` component passes the array of sources to the `picture` and lets - I guess 🤷 - the browser work out sizes.

This:

```
const forSource: (source: PictureSource) => string = (source) =>
    ` <source media="${mq(source)}" sizes="${source.width}px" srcset="${
        source.srcset
    }" />`;

export const Picture: React.FC<{
    sources: PictureSource[];
    alt: string;
    src: string;
    height: string;
    width: string;
    isLazy?: boolean;
}> = ({ sources, alt, src, height, width, isLazy = true }) => {
    return (
        // https://stackoverflow.com/questions/10844205/html-5-strange-img-always-adds-3px-margin-at-bottom
        // why did we put `style="vertical-align: middle;"` inside the img tag
        <picture
            dangerouslySetInnerHTML={{
                __html: `<!--[if IE 9]><video style="display: none;"><![endif]-->${sources
                    .map(forSource)
                    .join(
                        '',
                    )}<!--[if IE 9]></video><![endif]--><img style="vertical-align: middle;" itemprop="contentUrl" alt="${alt}" src="${src}" height="${height}" width="${width}" ${
                    isLazy ? 'loading="lazy"' : ''
                } />`,
            }}
        />
    );
};
```

Could we do something similar here so we don't have to have the rerender from useEffect?

Reply:
Not sure if we still need the IE9 workaround though 🤔

Reply:
+1, I think it's something we should avoid doing client side if possible!

Reply:
Totally agree with @gtrufitt, I have always been reluctant to be doing this client side. The only issue is that if we have a list of candidate overlay images with different widths, there is no way not to have logic for the best match.

We could however lift out the Overlay logic out of the youtube atom package and into DCR, that would allow us to tinker with the logic without having to do package updates. But fundamentally I do not see any easy work around.

The examples used here https://github.com/guardian/dotcom-rendering/pull/2040/files#diff-58a048c4d236ea52a26db21d08e39d97e2774b1e9547884b6b519034133fe0d0R3-R24 actually come from real JSON examples

Reply:
This is what srcset is for though I think? The image array that you get for a poster is in the same (I think) format as any other images, it seems that we should probably be passing the return value of the `Picture` component (given that array) to the atom and using that rather than having a background image in the atom? You'd need to server render that image, but I think that's the correct approach anyway? 

It's a bit gnarly and I don't 100% understand it so happy to have a chat / mob with the three of us on Monday though!